### DISCUSSION TITLE
Experimenting with other LLMs

### DISCUSSION THREAD

**Discussion by AshishSardana** [mayooear/ai-pdf-chatbot-langchain#74] [Q&A]
LangchainJS provides `HuggingFaceInference` class that can use the LLMs from HF Hub.

The documentation for using it is [here](https://js.langchain.com/docs/modules/models/llms/integrations) and the source code [here](https://github.com/hwchase17/langchainjs/blob/main/langchain/src/llms/hf.ts).

Has anyone tried replacing the `OpenAIChat` LLM with one of HuggingFace's LLM? 

I've tried by making simple changes to [makeChain.ts](https://github.com/mayooear/gpt4-pdf-chatbot-langchain/blob/main/utils/makechain.ts) and adding `HUGGINGFACE_HUB_API_KEY` in the `.env` file, but I'm receiving empty response from the chatbot.

**Comment by AshishSardana** [comment]
@mayooear - have you had a chance to try other LLMs? It'd be great if you can give it a quick try and share the necessary changes required.
I'd love to add the functionality where the user could select the LLM from the GUI in order to compare the response generated by them. For starters, the embeddings could be used from OpenAI's LLM but the response completion LLM can be dynamic.

**Reply by BaltekLabs** [reply]
Now with the ability to run models like llama and alpaca locally, would be nice to pull from local models like with [oobabooga](https://github.com/oobabooga/text-generation-webui)
